lib/models/per-bundle-addon-cache/addon-proxy.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+/**
+ * Returns a proxy to a target with specific handling for the
+ * `parent` property, as well has to handle the `app` property;
+ * that is, the proxy should maintain correct local state in
+ * closure scope for the `app` property if it happens to be set
+ * by `ember-cli`. Other than `parent` & `app`, this function also
+ * proxies _almost_ everything to `target.instance` with a few
+ * exceptions: we trap & return `[]` for `addons`, and we don't return
+ * the original `included` (it's already called on the "real" addon
+ * by `ember-cli`).
+ *
+ * Note: the target is NOT the singleton instance of the addon. Rather,
+ * it is a CacheEntry from SingletonAddonCache.
+ *
+ * @method getAddonProxy
+ * @param targetCacheEntry the PerBundleAddonCache CacheEntry we are to proxy
+ * @param parent the parent object of the proxy being created (the same as
+ * the 'parent' property of a normal addon instance)
+ * @return Proxy
+ */
+function getAddonProxy(targetCacheEntry, parent) {
+  let _app;
+  let addonProxy = new Proxy(targetCacheEntry, {
+    get(targetCacheEntry, property) {
+      if (property === 'parent') {
+        return parent;
+      }
+
+      if (property === 'app') {
+        return _app;
+      }
+
+      if (property === 'isProxy') {
+        return true;
+      }
```

Comment:
Who / what checks this property?

Reply:
removed this. I do have an equivalent now to allow references to the Symbol('_targetInstance_'), which is what now replaces the enumerable member 'targetInstance'.